### PR TITLE
feat: Native-image metadata for all modules

### PR DESCRIPTION
**Pull request by johanandren**
Third try is the charm - this is using classpath scanning during tests to find Akka extension points (extensions, serializers) that may be initialized through reflection and listing those for each module, in addition being able to manually add reflective accesses per module (I have not worked through that for each module yet)

### CODE REVIEW

File: akka-actor-tests/src/test/scala/akka/NativeImageMetadataSpec.scala [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      classOf[akka.actor.ActorCell].getName,
+      fields = Seq(
+        ReflectField("akka$actor$dungeon$Children$$_childrenRefsDoNotCallMeDirectly"),
+        ReflectField("akka$actor$dungeon$Children$$_functionRefsDoNotCallMeDirectly"),
+        ReflectField("akka$actor$dungeon$Children$$_nextNameDoNotCallMeDirectly"),
+        ReflectField("akka$actor$dungeon$Dispatch$$_mailboxDoNotCallMeDirectly"))),
+    ReflectConfigEntry(
+      classOf[akka.actor.RepointableRef].getName,
+      fields = Seq(ReflectField("_cellDoNotCallMeDirectly"), ReflectField("_lookupDoNotCallMeDirectly"))),
+    // FIXME replace reflective props and drop these
```
Comment (johanandren) [inline akka-actor-tests/src/test/scala/akka/NativeImageMetadataSpec.scala:30]: In progress in #32306

Comment (johanandren) [reply akka-actor-tests/src/test/scala/akka/NativeImageMetadataSpec.scala:30]: Actually, these two are probably not props-by-reflection but config based reflection so will stay here.

File: native-image-tests/local-scala/src/main/scala/com/lightbend/AkkaQuickstart.scala [hunk 2]
```diff
@@ -0,0 +1,67 @@
+package com.lightbend
+
+import akka.actor.typed.ActorRef
+import akka.actor.typed.ActorSystem
+import akka.actor.typed.Behavior
+import akka.actor.typed.scaladsl.Behaviors
+import com.lightbend.GreeterMain.SayHello
+
+object Greeter {
```
Comment (patriknw) [inline native-image-tests/local-scala/src/main/scala/com/lightbend/AkkaQuickstart.scala:9]: What is the plan for these tests? Do we add something that exercise more of cluster features? Do we need 4? local-scala, local-java, cluster-scala, cluster-java? Then we probably need something in Akka Projections that covers "everything"? (the existing sample there might do that?)

**Review by patriknw — COMMENTED** [review]
looking good

**Review by patriknw — COMMENTED** [review]
looking good

File: native-image-tests/local-scala/src/main/scala/com/lightbend/AkkaQuickstart.scala (hunk 2 above)
Comment (johanandren) [reply native-image-tests/local-scala/src/main/scala/com/lightbend/AkkaQuickstart.scala]: Yes, that was my thinking, to detect breakage as early and close to the broken thing as possible, we could possibly skip Java and exercise Java things from the Scala ones (extensions implemented in Java and whatever else is slightly different).

There will need to be follow up work in all Akka umbrella where we want to explicitly say native-image is supported. We may want to scope it to not cover all Akka umbrella, say we include HTTP, gRPC, R2DBC, k8 management and projections.

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**